DP/primitiveCalculater.cpp

```diff
@@ -0,0 +1,31 @@
+#include <iostream>
+#include <vector>
+#include <algorithm>
+
+using std::vector;
+
+vector<int> optimal_sequence(int n) {
+  std::vector<int> sequence;
+  while (n >= 1) {
+    sequence.push_back(n);
+    if (n % 3 == 0) {
+      n /= 3;
+    } else if (n % 2 == 0) {
+      n /= 2;
+    } else {
+      n = n - 1;
+    }
+  }
+  reverse(sequence.begin(), sequence.end());
+  return sequence;
+}
+
+int main() {
+  int n;
```

Comment:
below this line display a suitable message regarding what you are taking input from the user, for better user-interface